tests/Functional/PersistenceRepositoryTest.php

```diff
@@ -15,9 +15,15 @@ class PersistenceRepositoryTest extends WebTestCase
 {
     public function testRepositoryShouldBeSetCorrectly()
     {
-        static::bootKernel(['test_case' => 'ORM']);
+        self::bootKernel(['test_case' => 'ORM']);
 
-        $repository = static::$kernel->getContainer()->get('fos_elastica.manager.orm')
+        // returns the real and unchanged service container
+        $container = self::$kernel->getContainer();
+
+        // gets the special container that allows fetching private services
+        $container = self::$container;
```

Comment:
Can we add a debug in the test to see what's the class of this object?

Reply:
Yes something weird is going on. I'm doing just the same as the cookbook: https://symfony.com/blog/new-in-symfony-4-1-simpler-service-testing yet the test keeps failing.

Reply:
I'm getting `Symfony\Bundle\FrameworkBundle\Test\TestContainer`

Reply:
```
Keep in mind that, because of how Symfony's service container work, unused services are removed from the container. This means that if you have a private service not used by any other service, Symfony will remove it and you won't be able to get it as explained in this article. The solution is to define the service as public explicitly so Symfony doesn't remove it.
```

Reply:
But I'm using the service in the test :P

Reply:
Yes but the container is already compiled when trying to retrieve the service.